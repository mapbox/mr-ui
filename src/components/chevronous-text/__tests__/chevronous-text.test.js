import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './chevronous-text-test-cases';

describe('ChevronousText', () => {
  describe(testCases.basic.description, () => {
    const wrapper = shallow(
      React.createElement(testCases.basic.component, testCases.basic.props)
    );
    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.oneLongWord.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.oneLongWord.component,
        testCases.oneLongWord.props
      )
    );
    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.multilineText.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.multilineText.component,
        testCases.multilineText.props
      )
    );
    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.sizedIcon.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.sizedIcon.component,
        testCases.sizedIcon.props
      )
    );
    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.iconBeforeOneLongWord.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.iconBeforeOneLongWord.component,
        testCases.iconBeforeOneLongWord.props
      )
    );
    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.iconBeforeMultilineText.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.iconBeforeMultilineText.component,
        testCases.iconBeforeMultilineText.props
      )
    );
    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.sizedIconBeforeText.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.sizedIconBeforeText.component,
        testCases.sizedIconBeforeText.props
      )
    );
    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
