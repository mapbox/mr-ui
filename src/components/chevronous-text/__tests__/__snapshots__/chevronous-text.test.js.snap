// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronousText additional icon classes renders 1`] = `
<span>
  Explore
   
  <span
    className="txt-nowrap"
  >
    Mapbox
    <Icon
      inline={true}
      name="chevron-right"
      passthroughProps={Object {}}
      size={18}
      themeIcon="color-blue"
    />
  </span>
</span>
`;

exports[`ChevronousText basic renders 1`] = `
<span>
  Explore
   
  <span
    className="txt-nowrap"
  >
    Mapbox
    <Icon
      inline={true}
      name="chevron-right"
      passthroughProps={Object {}}
      size={18}
    />
  </span>
</span>
`;
